tests/std/tests/Dev11_0836436_get_time/test.cpp

```diff
@@ -799,13 +801,13 @@ void test_invalid_argument() {
     {
         wstringstream wss;
         wss << put_time(&currentTime, L"%Y-%m-%d-%H-%M");
-        assert(wss.rdstate() == ios_base::badbit);
+        assert(wss.rdstate() == ios_base::failbit);

```

Comment:
I'm concerned about this behavior change and can't find justification for it in the Standard. Please see my comment at https://github.com/microsoft/STL/issues/4882#issuecomment-2276788199 .

Note that this section of the test is calling `_set_invalid_parameter_handler(test_invalid_parameter_handler)` may have been incorrect to start with.

Reply:
Yep, I've also replied on the issue how I interpret the things.

Also you are right, `_set_invalid_parameter_handler` is no longer needed here as I validate the struct before letting `strftime` to crash or call the handler if set(as is the case here).

Reply:
Also, if "store unspecified characters and succeed" is determined to be the right approach, then instead of clamping values to zero (or maximum), writing the "expected" number of characters but using a fallback like `'*'` might be ideal, since it'll indicate that something was wrong with the input (e.g. `"2024-08-**"`).